### PR TITLE
feat: add an ability to manage user profile in settings

### PR DESCRIPTION
**Pull request by DeniGolf**

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   "devDependencies": {
     "@types/cors": "2.8.12",
     "@types/cryptr": "4.0.1",
+    "@types/multer": "^1.4.7",
```
Comment (AlikRakhmonov) [inline backend/package.json:29]:
```suggestion
    "@types/multer": "1.4.7",
```

File: backend/src/api/routes/index.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import userRoute from './user-route';
 import authRoute from './auth-route';
 
 const routes = (app: Express): void => {
-  app.use('/api/users', userRoute);
+  app.use('/api/user', userRoute);
```
Comment (AlikRakhmonov) [inline backend/src/api/routes/index.ts:6]: /users is correct

File: backend/src/common/helpers/multer.helper.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  private fileFilter(
+    _req: Request,
+    file: Express.Multer.File,
+    cb: (arg0: Error, arg1: boolean) => void,
+  ): void {
+    if (
+      file.mimetype === 'image/jpeg' ||
+      file.mimetype === 'image/png' ||
+      file.mimetype === 'image/gif'
```
Comment (AlikRakhmonov) [inline backend/src/common/helpers/multer.helper.ts:33]: ['image/jpeg', 'image/png',  'image/gif'].includes(file.mimetype)

File: backend/src/common/helpers/multer.helper.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import multer from 'multer';
+import { Request } from 'express';
+import path from 'path';
+import util from 'util';
+import fs from 'fs';
+
+class Multer {
```
Comment (AlikRakhmonov) [inline backend/src/common/helpers/multer.helper.ts:7]: why do you use classes? it's unnecessary

File: backend/src/common/helpers/s3-file-storage.helper.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import fs from 'fs';
+import S3 from 'aws-sdk/clients/s3';
+import { getOsEnv } from './path.helper';
+
+class S3FileStorage {
```
Comment (AlikRakhmonov) [inline backend/src/common/helpers/s3-file-storage.helper.ts:5]: why do you use classes? it's unnecessary

File: backend/src/data/entities/team.ts [hunk 6]
```diff
@@ -18,7 +18,7 @@ export class Team extends AbstractEntity {
   @Column()
   readonly workspaceId: string;
 
-  @ManyToOne(() => Workspace, workspace => workspace.teams)
+  @ManyToOne(() => Workspace, (workspace) => workspace.teams)
```
Comment (AlikRakhmonov) [inline backend/src/data/entities/team.ts:21]:
```suggestion
  @ManyToOne(() => Workspace, workspace => workspace.teams)
```

File: backend/src/data/entities/user.ts [hunk 7]
```diff
@@ -24,24 +19,27 @@ export class User extends AbstractEntity {
...
+
+  @OneToMany(() => RefreshToken, (RefreshToken) => RefreshToken.user)
   refreshTokens: RefreshToken[];
 
-  @OneToMany(() => UserWorkspace, userWorkspace => userWorkspace.user)
+  @OneToMany(() => UserWorkspace, (userWorkspace) => userWorkspace.user)
   userWorkspaces!: UserWorkspace[];
 
-  @ManyToMany(() => Team, team => team.users)
+  @ManyToMany(() => Team, (team) => team.users)
```
Comment (AlikRakhmonov) [inline backend/src/data/entities/user.ts:31]: remove all added brackets

File: backend/src/data/entities/user.ts [hunk 8]
```diff
@@ -24,24 +19,27 @@ export class User extends AbstractEntity {
   @Column({ length: 200, nullable: true })
   password: string;
 
-  @OneToMany(() => RefreshToken, RefreshToken => RefreshToken.user)
+  @Column({ length: 100, nullable: true })
```
Comment (AlikRakhmonov) [inline backend/src/data/entities/user.ts:22]:
```suggestion
  @Column({ nullable: true })
```

File: backend/src/server.ts [hunk 9]
```diff
@@ -14,8 +14,8 @@ const app: Express = express();
 
 app.use(cors());
 app.use(express.static(path.join(__dirname, '../../build', '/public')));
-app.use(express.json());
-app.use(express.urlencoded({ extended: true }));
+app.use(express.json({ limit: '50mb' }));
```
Comment (AlikRakhmonov) [inline backend/src/server.ts:17]: if you check limit for avatars, do it in its service method, not here

File: backend/src/services/user.service.ts [hunk 10]
```diff
@@ -1 +1,47 @@
-export const getSomething = <T>(data: T): Promise<T> => Promise.resolve(data);
+import UserRepository from '../data/repositories/user.repository';
+import { getCustomRepository } from 'typeorm';
+// import { HttpError } from '../common/errors/http-error';
+// import { HttpCode } from '../common/enums/http-code';
+// import { HttpErrorMessage } from '../common/enums/http-error-message';
```
Comment (AlikRakhmonov) [inline backend/src/services/user.service.ts:5]: remove commented lines

File: backend/src/data/migrations/1627557841360-add-tables.ts [hunk 11]
```diff
@@ -90,5 +214,4 @@ export class addTables1627557841360 implements MigrationInterface {
     await queryRunner.query('DROP TYPE "team_permission_option_enum"');
     await queryRunner.query('DROP TABLE "refresh_token"');
   }
-
 }
```
Comment (AlikRakhmonov) [inline backend/src/data/migrations/1627557841360-add-tables.ts:217]: this file should not be changed

File: backend/src/server.ts [hunk 12]
```diff
@@ -14,8 +14,8 @@ const app: Express = express();
...
+app.use(express.urlencoded({ extended: true, limit: '50mb' }));
```
Comment (AlikRakhmonov) [inline backend/src/server.ts:18]: the same

**Review by AlikRakhmonov — CHANGES_REQUESTED** [review]

File: backend/src/common/helpers/multer.helper.ts [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  unlinkFile(filePath: string): any {
+    return util.promisify(fs.unlink)(filePath);
+  }
+
+  private fileFilter(
+    _req: Request,
+    file: Express.Multer.File,
+    cb: (arg0: Error, arg1: boolean) => void,
```
Comment (aleksandr-kuharenko) [inline backend/src/common/helpers/multer.helper.ts:28]:
```suggestion
    cb: (err: Error, result: boolean) => void,
```

File: backend/src/common/helpers/s3-file-storage.helper.ts [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  _s3: S3;
+
+  constructor() {
+    const accessKeyId = getOsEnv('AWS_ACCESS_KEY');
+    const secretAccessKey = getOsEnv('AWS_SECRET_KEY');
```
Comment (aleksandr-kuharenko) [inline backend/src/common/helpers/s3-file-storage.helper.ts:10]: it should be in `.env.ts` file. Here should be:
const accessKeyId = env.s3.accessKey

**Comment by aleksandr-kuharenko** [comment]
Update `.ls-lint.yml` file to support filename with number(`s3-file-storage.helper.ts`)

>   .ts: regex:([.a-z\d]*)([-.][a-z\d]+)*

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]
Update `.ls-lint.yml` file to support filename with number(`s3-file-storage.helper.ts`)

>   .ts: regex:([.a-z\d]*)([-.][a-z\d]+)*

**Review by aleksandr-kuharenko — COMMENTED** [review]
Not fixed: https://github.com/BinaryStudioAcademy/bsa-2021-infostack/pull/13/files#r680777840

**Review by aleksandr-kuharenko — DISMISSED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

**Pull request merged**